src/tracebox/tracebox.cc

```diff
@@ -250,6 +229,85 @@ int TraceboxMain(int argc, char** argv) {
   return 0;
 }
 
+int TraceboxMain(int argc, char** argv) {
+  // Applet mode: invoke directly if argv[0] or argv[1] matches an applet name.
+  char* slash = strrchr(argv[0], '/');
+  char* argv0 = slash ? slash + 1 : argv[0];
+
+  for (const Applet& applet : g_applets) {
+    if (strcmp(argv0, applet.name) == 0)
+      return applet.entrypoint(argc, argv);
+    if (argc > 1 && strcmp(argv[1], applet.name) == 0)
+      return applet.entrypoint(argc - 1, &argv[1]);
+  }
+
+  if (argc <= 1) {
+    PrintTraceboxUsage();
+    return 1;
+  }
+
+  bool autodaemonize = false;
+  bool use_system_sockets = false;
+  for (int i = 1; i < argc;) {
+    if (strcmp(argv[i], "--autodaemonize") == 0) {
+      autodaemonize = true;
+      memmove(static_cast<void*>(&argv[i]), static_cast<void*>(&argv[i + 1]),
+              sizeof(char*) * static_cast<size_t>(argc - i - 1));
+      argc--;
+    } else {
+      if (strcmp(argv[i], "--system-sockets") == 0) {
+        use_system_sockets = true;
+      }
+      ++i;
+    }
+  }
+
+  if (autodaemonize) {
+    if (use_system_sockets) {
+      PERFETTO_ELOG(
+          "Warning: --system-sockets with --autodaemonize is supported but "
+          "not recommended. Prefer `tracebox ctl start` for persistent system "
+          "sockets.");
+    }
+    return RunAutodaemonize(argc, argv);
+  }
+
+  if (use_system_sockets) {
+    PERFETTO_FATAL(
```

Comment:
Fatal seems a bit aggressive given that this is just the new default?
I would

1) Move this message in the for loop above. there i sno need to set a variable, just to emit a lot here:
2) SOften this into a PERFETTO_ELOG ?